packages/headless/src/controllers/relevance-inspector/headless-relevance-inspector.ts

```diff
@@ -159,6 +160,12 @@ export function buildRelevanceInspector(
             this.logInformation();
           }
           listener();
+          wasEnabled = this.state.isEnabled;
+          return;
+        }
+        if (this.state.isEnabled !== wasEnabled) {
```

Comment:
I'm not sure I understand what was the bug before ?

Also, I think the logic should be that it calls subscribers when there is a change in the "global" state, and not a local variable like `wasEnabled` that you added here.

Meaning, If i have two separate relevance inspector instance, if I call enable on one, both should notify their listener, and vice versa for disable.

Reply:
The bug was that enabling or disabling a relevance inspector would not notify the listeners of any instance at all.

`wasEnabled` is only created for the purpose of checking whether `isEnabled` was changed.

Reply:
I feel this fix is too specific, and does not capture the deeper issue.

**Problem**

Our controllers generally call subscribers when the state of the controller changes. That doesn't happen here because the `subscribe` method was overridden. This custom `subscribe` only call it's subscribers only if the response search id changes, and not when client-side state changes occur (e.g. enable/disable). This is inconsistent with the rest of our controllers.

**Quick Solution**

We can reuse the core controller behaviour and compose it with the relevance inspector specific behaviour (i.e. logging to the console). We will need two subscriptions, and need to manage unsubscribing. e.g.

```js
const controller = buildController();

return {
    ...controller,
    subscribe(cb) {
        const unsubscribe1 = engine.subscribe(() => {
            ... custom logic to log on search id change ...
       })

        const unsubscribe2 = controller.subscribe(cb) // classic behaviour
        return () => {
            unsubscribe1();
            unsubscribe2();
        }
    }
}
```

**Better solution**

The issue with the above fix (and what we currently have) is logging happens at the controller level. If someone creates `n` relevance inspector controller instances, then assuming they are all configured the same, there will be `n` logs in the console.

Instead, we should place our state in the central store. All relevance inspector instances are either enabled or disabled. It should not be possible to have some in one state and some in another.

Moreover, logging would ideally be pushed down to the action layer (e.g. done inside the executeSearch action, provided `isEnabled` is `true`), so that only one log occurs when we get a new search response, regardless of the number of relevance inspector instances.